### PR TITLE
feat(amazonq): Add acknowledgement button for disclaimer

### PR DESCRIPTION
**Pull request by jpinkney-aws**
## Problem
- disclaimer is too wordy

## Solution
- go back to the old disclaimer and add an ackowledgement button

---

- Treat all work as PUBLIC. Private `feature/x` branches will not be squash-merged at release time.
- Your code changes must meet the guidelines in [CONTRIBUTING.md](https://github.com/aws/aws-toolkit-vscode/blob/master/CONTRIBUTING.md#guidelines).

License: I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/src/amazonq/webview/messages/messageDispatcher.ts [hunk 1]
```diff
@@ -60,12 +60,19 @@ export function dispatchWebViewMessagesToApps(
...
                     telemetry.ui_click.emit({
                         elementId: msg.source,
                         result: 'Succeeded',
                     })
+                    return
                 }
+                return
+            }
+            case 'disclaimer-acknowledged': {
+                void globals.globalState.tryUpdate('aws.amazonq.disclaimerAcknowledged', true)
```
Comment (justinmk3) [inline packages/core/src/amazonq/webview/messages/messageDispatcher.ts:74]: tryUpdate is synchronous
```suggestion
               globals.globalState.tryUpdate('aws.amazonq.disclaimerAcknowledged', true)
```

File: packages/core/src/amazonq/webview/ui/disclaimer/disclaimer.ts
Comment (justinmk3) [inline packages/core/src/amazonq/webview/ui/disclaimer/disclaimer.ts:1]: does this need its own subfolder?

File: packages/core/src/amazonq/webview/ui/main.ts [hunk 2]
```diff
@@ -609,6 +613,24 @@ export const createMynahUI = (
...
+
+                // create telemetry
+                ideApi.postMessage(createClickTelemetry('amazonq-disclaimer-acknowledge-button'))
+
+                // remove all disclaimer cards from all tabs
+                Object.keys(mynahUI.getAllTabs()).forEach((storeTabKey) => {
+                    // eslint-disable-next-line unicorn/no-null
+                    mynahUI.updateStore(storeTabKey, { promptInputStickyCard: null })
+                })
+            }
```
Comment (justinmk3) [inline packages/core/src/amazonq/webview/ui/main.ts:632]: should it end with `return` like the others or does it intentionally fall through (is that worth calling out in a comment?)

**Review by justinmk3 — APPROVED** [review]

**Review by dogusata — APPROVED** [review]

File: packages/core/src/amazonq/webview/ui/main.ts (hunk 2 above)
Comment (dogusata) [reply packages/core/src/amazonq/webview/ui/main.ts:632]: In general, a button can do more than one thing, but as it seems we don't have this kind of separation here. It can be converted to a switch statement.

**Review by mr-lee — APPROVED** [review]

**Review by justinmk3 — APPROVED** [review]
